app/src/main/contrib/components/Responsibilities/BurdenEstimates/UploadBurdenEstimatesForm.tsx

```diff
@@ -1,86 +1,345 @@
+        })
+    };
+
+    componentDidMount = () => {
+
+        const settings: ConfigurationHash = {
+            ...baseSettings,
+            target: (params) => `${this.props.url}?${params.join("&")}`,
+            minFileSizeErrorCallback: () => {
+                this.props.resetPopulateState();
+                this.setState({
+                    ...initialUploadState,
+                    metadata: {...this.state.metadata},
+                    uploadErrors: [{
+                        code: "e",
+                        message: "This file is empty."
+                    }]
+                })
+            }
+        };
+
+        this.uploadClient = this.props.createUploadClient(settings);
+        this.uploadClient.assignBrowse(this.uploaderElement, false);
+
+        this.uploadClient.on('fileAdded', (file: ResumableFile) => {
+
+            this.props.resetPopulateState();
+            const validationResult = checkFileExtensionIsCSV(file.fileName);
+
+            this.setState({
+                ...initialUploadState,
+                metadata: {...this.state.metadata},
+                fileValidationResult: validationResult,
+                file: file
+            });
+        });
+
+        this.uploadClient.on('progress', () => {
+
+            this.setState({
+                isUploading: true,
+                progress: roundToOneDecimalPlace(this.uploadClient.progress() * 100)
+            });
+        });
+
+        this.uploadClient.on('fileSuccess', () => {
+
+            this.resetUploadState(false);
+            this.props.createBurdenEstimateSet(this.state.metadata);
+            this.props.populateEstimateSet(this.props.uploadToken);
+        });
+
+        this.uploadClient.on('fileError', (file: ResumableFile, error: string) => {
+
+            let uploadErrors: ErrorInfo[] = [];
+            try {
+                const result = JSON.parse(error) as Result;
+                uploadErrors = result.errors
+            } catch (e) {
+                uploadErrors.push({code: "error", message: "Error contacting server"})
+            }
+            this.setState({
+                uploadErrors: uploadErrors,
+                isUploading: false,
+                progress: 0
+            });
+        });
+    };
+
+    startUpload = () => {
+
+        this.props.resetPopulateState();
+        this.props.createBurdenEstimateSet(this.state.metadata);
+
+        const tryUpload = setInterval(() => {
+            if (!this.uploadDisabled()) {
+
+                clearInterval(tryUpload);
+                this.setState({
+                    isUploading: true
+                });
+
+                this.uploadClient.upload();
+            }
+        });
+
+    };
+
+    cancelUpload = () => {
+        this.resetUploadState(true);
+        this.uploadClient.cancel();
+    };
+
+    uploadDisabled = () => {
+        return this.submitDisabled()
```

Comment:
We allow submission when all fields are filled in, but until the set has been created the url will be null (because the newly created set id forms part of the url) so we need to wait for the url to be non null before beginning file upload